File: python/structures/test/sorts/merge_sort_test.py
import structures.src.sorts.merge as merge_sort
import structures.src.util.constants as const
from structures.src.util.sort_util import is_sorted, random_array


def test_sorted():
    arr = random_array(const.ARRAY_SIZE)

    # make sure the starting array isn't sorted
    assert not is_sorted(arr)

    # sort the array
    result = merge_sort.merge_sort(arr)

    # make sure it worked
    assert is_sorted(result)

File: ruby/test/sorts/merge_sort_test.rb
require_relative '../../src/sorts/merge_sort'
require_relative '../../src/util/util'
require_relative '../../src/util/constants'
require_relative '../util'

require 'test_helper'
require 'minitest/autorun'
require 'minitest/spec'

describe MergeSort do
    describe 'when passed an array' do
        it 'should correctly sort it' do
            arr = TestUtil.random_array(Constants.ARRAY_SIZE)

            result = MergeSort.sort(arr)

            Util.is_sorted(result).must_equal true
        end
    end
end
